### PR TITLE
feat(charts): Add Truncation Bool Prop for Chart Legend Labels

### PR DESCRIPTION
**Pull request by ayesha-omarali**
Defaults to true; truncates labels. 
Pass in truncateLegend=false to not truncate.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/charts/baseChart.jsx [hunk 1]
```diff
@@ -54,6 +54,9 @@ class BaseChart extends React.Component {
     // Chart legend
     legend: SentryTypes.EChartsLegend,
 
+    // Chart legend truncate labels
+    legendTruncate: PropTypes.bool,
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/charts/baseChart.jsx]: hmmm, how about `truncateLegend` or something - I think making this a verb would be slightly better

Comment (ayesha-omarali) [reply src/sentry/static/sentry/app/components/charts/baseChart.jsx]: 👍

**Review by billyvg — APPROVED** [review]

**Comment by lynnagara** [comment]
I think we should align with how truncation was done in tooltips rather than have this different interface for truncating legends, i.e. truncateLegend shouldn't be a prop of baseChart

**Comment by lynnagara** [comment]
@billyvg What do you think? I'd be in favour of reverting this or at least aligning with tooltip

**Comment by billyvg** [comment]
Ah yeah I forgot we do this in tooltip, 👍 for aligning w/ tooltip

**Comment by ayesha-omarali** [comment]
Sounds good to me, will fix it!

**Pull request merged**